### ISSUE TITLE
ec2_group module not using array with rules parameter

### ISSUE THREAD

**Issue by windowsrefund** [ansible/ansible#14609]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.1.0 (devel e35b1cf154) last updated 2016/02/22 11:05:15 (GMT -400)
```
##### Ansible Configuration:

Nothing special
##### Environment:

N/A
##### Summary:

The rules parameter of the ec2_group module is not taking a list as its documentation suggests.
##### Steps To Reproduce:

Use data with the following schema:

```
defaults:
  security_groups:
    ssh:
      rules:
        - proto: tcp
          to_port: 22
          from_port: 22
          cidr_ip: 0.0.0.0/0
```

Here's the play along with debug info showing that the data is properly formatted

```
      - debug: var=item.value                                                  
        with_dict: "{{ defaults.security_groups }}"

     - name: create security groups                                            
        ec2_group:                                                              
          aws_access_key: "{{ lookup('env', 'AWS_ACCESS_KEY_ID') }}"            
          aws_secret_key: "{{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}"        
          state: present                                                        
          name: "{{ item.key }}"                                                
          description: "{{ item.value.description | default(item.key) }}"      
          region: "{{ defaults.region }}"                                      
          vpc_id: "{{ vpc_info.vpcs[0].id }}"                                  
          rules: "{{ item.value.rules }}"                                      
        with_dict: "{{ defaults.security_groups }}"

```
##### Expected Results:

The ec2_group module should have created the security group.
##### Actual Results:

It blew up. Here's the debug message and error.

```
TASK [debug] *******************************************************************
ok: [localhost] => (item={'value': {u'rules': [{u'to_port': 22, u'from_port': 22, u'cidr_ip': u'0.0.0.0/0', u'proto': u'tcp'}]}, 'key': u'ssh'}) => {
    "item": {
        "key": "ssh",
        "value": {
            "rules": [
                {
                    "cidr_ip": "0.0.0.0/0",
                    "from_port": 22,
                    "proto": "tcp",
                    "to_port": 22
                }
            ]
        }
    },
    "item.value": {
        "rules": [
            {
                "cidr_ip": "0.0.0.0/0",
                "from_port": 22,
                "proto": "tcp",
                "to_port": 22
            }
        ]
    }
}

TASK [create security groups] **************************************************

    failed: [localhost] => (item={'value': {u'rules': [{u'to_port': 22, u'from_port': 22, u'cidr_ip': u'0.0.0.0/0', u'proto': u'tcp'}]}, 'key': u'ssh'}) => {"failed": true, "item": {"key": "ssh", "value": {"rules": [{"cidr_ip": "0.0.0.0/0", "from_port": 22, "proto": "tcp", "to_port": 22}]}}, "msg": "Invalid rule parameter '['"}


```

**Comment by kbroughton** [comment]
I'm seeing the same.  The error popped up between 2.0.0.2 and 2.1.0
I can confirm the fix #14609 works for me.

**Comment by bcoca** [comment]
# Module Repo Information

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a submission about a module, and aside from action_plugins, if you know what those are, the modules
in ansible are now moved to separate repos.  We would appreciate if you can submit this there instead.

If this is about a new module, submit pull requests or ideas to:
- https://github.com/ansible/ansible-modules-extras

If this is about an existing module:
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your module to https://github.com/ansible/ansible-modules-core

Additionally, should you need more help with this, you can ask questions on:
- IRC: #ansible on irc.freenode.net  
- mailing list: https://groups.google.com/forum/#!forum/ansible-project

Thanks!
